src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -770,11 +801,40 @@ public boolean hasTownHall()
         return townHall != null;
     }
 
+
     /**
-     * Get building in Colony by ID.
+     * Get field in Colony by ID
      *
-     * @param buildingId ID (coordinates) of the building to get.
-     * @return AbstractBuilding belonging to the given ID.
+     * @param fieldId    ID (coordinates) of the field to get
+     * @return              Fuilding belonging to the given ID
+     */
+    public Field getField(BlockPos fieldId)
+    {
+        return fields.get(fieldId);
+    }
+
+    /**
+     * Returns a field which has not been taken yet.
+     * @return a field if there is one available, else null.
+     */
+    public Field getFreeField()
+    {
+        for(Field field: fields.values())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)